### ISSUE TITLE
puppeteer-firefox: ERROR: failed to call method 'Browser.deleteCookies'

### ISSUE THREAD

**Issue by brandon-bethke-timu** [puppeteer/puppeteer#4470]
I receive the following error when I try to run the following example using puppeteer-firefox version 0.5.0:
```
const pptrFirefox = require('puppeteer-firefox');
(async () => {
  const browser = await pptrFirefox.launch({headless: false});
  const page = await browser.newPage();
  await page.goto('https://www.google.com');
  await page.setCookie({name: 'mycookie', value: 'myvalue'});
})();

```

```
$ node sample.js 
(node:92226) UnhandledPromiseRejectionWarning: Error: Protocol error (Browser.deleteCookies): ERROR: failed to call method 'Browser.deleteCookies' with parameters {
  "browserContextId": null,
  "cookies": [
    {
      "url": "https://www.google.com/",
      "name": "mycookie"
    }
  ]
}
Expected "<root>.browserContextId" to be |string|; found |object| `null` instead. _dispatch@chrome://juggler/content/protocol/Dispatcher.js:100:15

    at node_modules/puppeteer-firefox/lib/Connection.js:74:56
    at new Promise (<anonymous>)
    at Connection.send (node_modules/puppeteer-firefox/lib/Connection.js:73:12)
    at Page.deleteCookie (node_modules/puppeteer-firefox/lib/Page.js:115:22)
    at Page.<anonymous> (node_modules/puppeteer-firefox/lib/helper.js:32:23)
    at Page.setCookie (node_modules/puppeteer-firefox/lib/Page.js:135:16)
    at Page.<anonymous> (node_modules/puppeteer-firefox/lib/helper.js:32:23)
    at sample.js:6:14

```

**Comment by aslushnikov** [comment]
@brandon-bethke-neudesic thank you for taking time to file this!

**Comment by michaelcypher** [comment]
I'm seeing the exact same issue for `Browser.getCookies` for puppeteer-firefox version 0.5.0.

```js
const pptrFirefox = require('puppeteer-firefox');
(async () => {
  const browser = await pptrFirefox.launch({headless: false});
  const page = await browser.newPage();
  await page.goto('https://www.google.com');
  const cookies = await page.cookies(page.url());
})();
```

```
node:36006) UnhandledPromiseRejectionWarning: Error: Protocol error (Browser.getCookies): ERROR: failed to call method 'Browser.getCookies' with parameters {
  "browserContextId": null,
  "urls": [
    "https://www.google.com/"
  ]
}
Expected "<root>.browserContextId" to be |string|; found |object| `null` instead. _dispatch@chrome://juggler/content/protocol/Dispatcher.js:100:15

    at Promise (/Users/mcypher/plaid/pdaas/node_modules/puppeteer-firefox/lib/Connection.js:74:56)
    at new Promise (<anonymous>)
    at Connection.send (/Users/mcypher/plaid/pdaas/node_modules/puppeteer-firefox/lib/Connection.js:73:12)
    at Page.cookies (/Users/mcypher/plaid/pdaas/node_modules/puppeteer-firefox/lib/Page.js:90:30)
    at Page.<anonymous> (/Users/mcypher/plaid/pdaas/node_modules/puppeteer-firefox/lib/helper.js:32:23)
    at /Users/mcypher/plaid/pdaas/sample.js:6:30
```

**Comment by aslushnikov** [comment]
@mikeecb thanks! We're working on the fix.

The workaround would be to use incognito browsing contexts for now - cookies should work just fine for these.

**Comment by james-berry** [comment]
still getting ERROR: failed to call method 'Browser.getCookies' using:

```
const browser = await puppeteer.launch({
    headless: false,
    args: ['--start-fullscreen', '--incognito'],
    devtools: true,
  });
```

**Comment by nayanamana** [comment]
Have anyone found workarounds for this issue?

**Comment by whimboo** [comment]
@mathiasbynens I assume we can close this issue given that it is about Juggler?

**Comment by mjzffr** [comment]
I'm closing this issue since the experimental `puppeteer-firefox` package is no longer maintained. A replacement will be available
later this year. You can learn more about Mozilla's work on this at https://wiki.mozilla.org/Remote

See also https://github.com/puppeteer/puppeteer/pull/5137

**Comment by limonte** [comment]
FWIW, this issue has been fixed in #4850 and released in `puppeteer-firefox@0.5.1` which was published 4 days ago.
